java10/session-1-Local-Variable-Type-Inference/04_Full_uses/src/com/typeinference/Java10var.java

```diff
@@ -0,0 +1,147 @@
+package com.typeinference;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+
+/*
+* This class has some variable declarations that can't be changed for Java 10 type inference way,
+* you should try to change whichever you think can use var, and then compile to check if your changes are correct
+* */
+public class Java10var {
+
+	public ZoneId zoneIdAraucaColombia = ZoneId.of( "GMT-5" );
+	public ZoneId zoneIdIstanbulTurkey = ZoneId.of( "GMT+3" );
+	public ZoneId zoneIdLondonUK = ZoneId.of( "GMT+1" );
+
+	public static void main(String args[]) {
+
+		Java10var java10var = new Java10var();
+		List<Long> numberss = List.of( 13L, 11L, 12L, 13L, 10L, 15L, 12L, 11L, 10L );
+
+		Double average = java10var.average( numberss );
+		System.out.println( "Average is: " + average );
+
+		Long first = java10var.getFirst( numberss );
+		System.out.println( "The first number in the list is: " + first );
+
+		java10var.printTimes();
+
+		printMonth();
+
+		printNumbers();
+	}
+
+	public Double average(List<Long> numberss) {
+
+		Double average = 0D;
+		Double sum = numberss.stream().mapToDouble( p -> p ).sum();
+		average = sum / numberss.size();
+
+		return average;
+	}
+
+	public Long getFirst(List<Long> numberss) {
+		Long first;
+		first = numberss.get( 0 );
+		return first;
+	}
+
+
+	public void printTimes() {
+		LocalDateTime localDateTimeAraucaColombia = LocalDateTime.now( zoneIdAraucaColombia );
+		LocalDateTime localDateTimeIstanbulTurkey = LocalDateTime.now( zoneIdIstanbulTurkey );
+		LocalDateTime localDateTimeLondoUK = LocalDateTime.now( zoneIdLondonUK );
+
+		System.out.println( "Current time in Arauca - Colombia is: " + localDateTimeAraucaColombia );
+		System.out.println( "Current time in Istanbul - Turkey is: " + localDateTimeIstanbulTurkey );
+		System.out.println( "Current time in London - United Kingdom is: " + localDateTimeLondoUK );
+
+	}
+
+	public static void printMonth() {
+
+
+		int month = LocalDate.now().getMonthValue();
+		String monthString;
+		switch ( month ) {
+			case 1:
```

Comment:
You can shorten this piece of code by using a map or an array, makes it better readable.

Reply:
@neomatrix369 the code of the examples are not the best code,  in this case I wanted users try to use var in a method that has a switch. I guess you meant I should change the switch to a map?

I'm gonna add a method with Map, it can be interested figure out if var works for the Key

Reply:
Please apply that, switch isn't a good practise so best not to encourage developers to do that

Reply:
Can we use any factory method like `Array.asList(...)` and then convert it into a `LinkedList` ? Smaller constructs are good as the focus is on `var` not on other ceremonies, See lines 117 onwards